### DISCUSSION TITLE
Access to Android android:hint

### DISCUSSION THREAD

**Discussion by v-mwalk** [appium/appium#16484] [Q&A] [answered]
This is quite possibly a UIAutomator2 thing; but thought I'd ask here....

In the AUT I'm working with, `android.widget.EditText` elements (com.google.android.material.textfield.TextInputEditText) are labeled using the `android:hint` properties.

I am trying to create a Selector that selects the element using that 'hint'.  However, the hint property does not appear to be being exposed - Appium Inspector cant cant see the 'hint' and it doesn't appear to be exposed as another name.

Is there a, possibly obscure, attribute name I can use to get at the 'hint'?  Either in the Selector directly for from a .getAttribute call?

**Comment by KazuCocoa** [comment, marked as answer]
Probably no for now.

https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#getHintText() it seems like the api is available since api level 26.
So, once we've added it in https://github.com/appium/appium-uiautomator2-server/blob/b185f317a0fa2a7fa3a59a1f466372083236d2af/app/src/main/java/io/appium/uiautomator2/model/UiObjectElement.java , then it should be available.

**Reply by KazuCocoa** [reply]
i haven't tested, but probably https://github.com/appium/appium-uiautomator2-server/pull/460 can do for api 26+ env.

**Reply by v-mwalk** [reply]
Thanks @KazuCocoa , will try to give it a go today; looking at the PR, looks cool.

Could I ask, if using XPath selector, where is the xpath interpretation done?  IE. If i had a selector `//android.widget.TextEdit[@hint="Hello world"]` where is the Xpath processed against the DOM (what builds the DOM XML that the xpath is applied to?)

**Reply by KazuCocoa** [reply]
I haven't tested it yet, but maybe can as `@hint`. or i need to update the PR in XPath as well.

**Reply by KazuCocoa** [reply]
I've tested https://github.com/appium/appium-uiautomator2-server/pull/460 could return `hint` as an attribute.
@v-mwalk Do you have a good example app to test `hint`? I've checked some apps in an emulator like Google tools, but it seems no `hint` attribute. I only tested `hint` attribute returns `nil`.

**Reply by KazuCocoa** [reply]
I've tested. Aa https://github.com/appium/appium-uiautomator2-server/pull/460#issuecomment-1050664637, XPath also worked.

**Reply by jlipps** [reply]
> Could I ask, if using XPath selector, where is the xpath interpretation done? IE. If i had a selector //android.widget.TextEdit[@hint="Hello world"] where is the Xpath processed against the DOM (what builds the DOM XML that the xpath is applied to?)

It is in the UiAutomator2 server. The entry point is here: https://github.com/appium/appium-uiautomator2-server/blob/master/app/src/main/java/io/appium/uiautomator2/handler/FindElement.java#L87-L92

**Comment by Idoserovitz** [comment]
Hi @KazuCocoa, when will it be available to use through appium?

**Reply by KazuCocoa** [reply]
Latest UIA2 driver has it, which requires Appium 2.

**Reply by Idoserovitz** [reply]
is the appium@next package stable and reliable enough to use?
is there a release date /  an estimation for appium 2?

**Reply by v-mwalk** [reply]
I'm using the Appium 2 beta in a commercial env and have had no issues at all; and I'm giving it a good thrashing.  Downside was having to explain to boss why I'm using a Beta package and that I'm happy to take the risks associated with a Beta - but apart from that, no tech issues at all.  I'd say Appium 2 Beta is pretty darn good.

**Reply by jlipps** [reply]
The only reason Appium 2 is still in beta is because it's taking forever to rewrite the docs and docs site and we don't want to publish Appium 2 officially until the docs are ready. At this point we're just continuing to develop Appium along the Appium 2 trajectory. It's not any less stable than any previous version of Appium. The only way we know if there are issues is (a) if our CI catches something, and (b) if people use it and tell us about it. We keep our CI green, so the more people use Appium 2, the more well-tested it will be. Managerial hesitation to use 'beta' software is all well and good but for open source projects the beta/non-beta distinction often doesn't make too much sense.

**Reply by jlipps** [reply]
And thanks for the report @v-mwalk!

**Reply by Idoserovitz** [reply]
Hi @jlipps, thanks for the detailed response. Great to hear and it will be my pleasure to test it along with you guys. Thanks again!
